README.md

```diff
@@ -1,11 +1,67 @@
-# mamba.np 
+# mamba.np
+
 `mamba.np` is a pure NumPy implementation of Mamba.
 
 <p align="center">
   <img src="assets/mamba.jpeg" width="300" alt="mamba.np">
 </p>
 
+## Prerequisites
+
+-   **Python**: This project requires Python version 3.8 or higher. It has been tested on Python 3.10.
+
+### Environment Setup
+
+1. **Clone the Repository**
+
+    First, clone the repository to your local machine:
+
+    ```bash
+    git clone git@github.com:idoh/mamba.np.git
+    cd mamba.np
+    ```
+
+2. **Set Up Virtual Environment**
+
+    It is recommended to use a virtual environment to manage dependencies. You can set up a virtual environment using `venv`:
+
+    ```bash
+    python3 -m venv venv
+    ```
+
+3. **Activate Virtual Environment**
 
+    - On **Windows**:
+        ```bash
+        .\venv\Scripts\activate
+        ```
+    - On **macOS/Linux**:
+        ```bash
+        source venv/bin/activate
+        ```
+
+4. **Install Required Packages**
+
+    Install the required packages using `pip`:
+
+    ```bash
+    pip install -r requirements.txt
+    ```
+
+5. **Install PyTorch**
+
+    Install PyTorch from the official [PyTorch website](https://pytorch.org/get-started/locally/). Choose the appropriate configuration for your system. For example, for a basic CPU-only installation on Windows/Mac, you can use:
+
+    ```bash
+    pip install torch
```

Comment:
Since we only use torch CPU, I believe that specifying it in the `requirements.txt` file is sufficient.